File: Android/app/src/test/java/com/nfsindustries/cryptocurrencymonitor/model/CurrencyListItemTest.java
package com.nfsindustries.cryptocurrencymonitor.model;

import junit.framework.Assert;

import org.junit.Test;

/**
 * Created by maxmo on 29/5/2017.
 */

public class CurrencyListItemTest {

    @Test
    public void shouldInitAndSet() throws Exception {
        CurrencyListItem cryptocurrency = new CurrencyListItem("bitcoin", "BTC");
        Assert.assertNotNull(cryptocurrency);
        Assert.assertNotNull(cryptocurrency.getName());
        Assert.assertNotNull(cryptocurrency.getSymbol());
        Assert.assertEquals(cryptocurrency.getName(), "bitcoin");
        Assert.assertEquals(cryptocurrency.getSymbol(), "BTC");
    }

    @Test
    public void shouldConvertToString() throws Exception {
        CurrencyListItem cryptocurrency = new CurrencyListItem("dogecoin", "DOGE");
        Assert.assertNotNull(cryptocurrency);
        Assert.assertNotNull(cryptocurrency.getName());
        Assert.assertNotNull(cryptocurrency.getSymbol());
        Assert.assertNotNull(cryptocurrency.toString());
    }
}

File: Android-Kotlin/app/src/test/java/com/nfsindustries/cryptocurrencymonitor/model/CurrencyListItemTest.kt
import com.nfsindustries.cryptocurrencymonitor.model.CurrencyListItem
import junit.framework.Assert
import org.junit.Test

/**
 * Created by MaxMouse on 25/9/2017.
 */

class CurrencyListItemTest {

    @Test
    @Throws(Exception::class)
    fun shouldInitAndSet() {
        val cryptocurrency = CurrencyListItem("bitcoin", "BTC")
        Assert.assertNotNull(cryptocurrency)
        Assert.assertNotNull(cryptocurrency.name)
        Assert.assertNotNull(cryptocurrency.symbol)
        Assert.assertEquals(cryptocurrency.name, "bitcoin")
        Assert.assertEquals(cryptocurrency.symbol, "BTC")
    }

    @Test
    @Throws(Exception::class)
    fun shouldConvertToString() {
        val cryptocurrency = CurrencyListItem("dogecoin", "DOGE")
        Assert.assertNotNull(cryptocurrency)
        Assert.assertNotNull(cryptocurrency.name)
        Assert.assertNotNull(cryptocurrency.symbol)
        Assert.assertNotNull(cryptocurrency.toString())
    }
}